pkg/flowkit/config/json/account_test.go

```diff
@@ -344,7 +344,7 @@ func Test_TransformDefaultAccountToJSON(t *testing.T) {
 	x, _ := json.Marshal(j)
 
 	// our output format is shorted - improve test
-	assert.NotEqual(t, string(b), string(x))
+	assert.Equal(t, string(b), string(x))
```

Comment:
Since we now set `SaveAdvancedFormat` to true after `transformAdvancedToConfig`, this will cause this test to be equal even if the json format has default values, if it appears as advanced format, i.e. `key` field being not a string

Reply:
I think this test is specifically meant to test the case where an account key with default values is transformed into the simplified JSON format, so I don't think we should change the expected value here.

I think the test can be improved, though (as the comment on line 346 suggests).

Rather than parsing the JSON, maybe just start with an `Account` object.

```go
// You can use a helper to create this account if you want to re-use it in other tests.
func defaultAccount(t *testing.T) config.Account {
  privateKey, err := crypto.DecodePrivateKeyHex(
    "1272967fd2bd75234ae9037dd4694c1f00baad63a10c35172bf65fbb8ad74b47",
    crypto.ECDSA_P256,
  )
  assert.NoError(t, err)

  key := flowkit.NewHexAccountKeyFromPrivateKey(0, crypto.SHA3_256, privateKey)

  account := config.
    NewAccount("emulator-account).
    SetAddress(flow.HexToAddress("f8d6e0586b0a20c7")).
    SetKey(key)

  return account
}

// ...

account := defaultAccount(t)

accounts := []Account{account}

j := transformAccountsToJSON(accounts)
result, err := json.Marshal(j)
assert.NoError(t, err)

expected := []byte(`{"emulator-account":{"address":"f8d6e0586b0a20c7","key":"1272967fd2bd75234ae9037dd4694c1f00baad63a10c35172bf65fbb8ad74b47"}}`)

assert.Equal(t, result, expected)
```